src/main/webapp/app/exercise/feedback/feedback.component.ts

```diff
@@ -234,18 +240,17 @@ export class FeedbackComponent implements OnInit, OnChanges {
                     if (
                         this.result.assessmentType !== AssessmentType.AUTOMATIC_ATHENA &&
                         this.exerciseType === ExerciseType.PROGRAMMING &&
-                        this.result.submission?.participation &&
-                        (this.result.submission as ProgrammingSubmission).buildFailed
+                        (this.participation?.submissions?.[0] as ProgrammingSubmission)?.buildFailed
                     ) {
-                        return this.fetchAndSetBuildLogs(this.result.submission.participation.id!, this.result.id);
+                        return this.fetchAndSetBuildLogs(this.participation.id!, this.result.id);
```

Comment:
We have never checked if `participation.id` is defined - or am I missing something? Can we please add a check before asserting that it is not undefined?